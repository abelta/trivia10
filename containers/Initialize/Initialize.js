import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-native';
import { Spinner, View } from '@shoutem/ui';
import { resetted as gameResetted } from '../../actions/game';
import { fetchSent as questionsFetchSent } from '../../actions/questions';
import { questionShape } from '../../propTypeShapes/questions';

class Initialize extends PureComponent {
  componentWillMount() {
    const { gameReset, questionsFetch } = this.props;
    gameReset();
    questionsFetch();
  }

  render() {
    const { questions, round } = this.props;
    if (round === 0 && questions.length > 0) return <Redirect to="/question" />;
    return (
      <View styleName="flexible" style={{ justifyContent: 'center' }}>
        <Spinner style={{ size: 'large' }} />
      </View>
    );
  }
}

Initialize.propTypes = {
  gameReset: PropTypes.func.isRequired,
  questions: PropTypes.arrayOf(PropTypes.shape(questionShape)).isRequired,
  questionsFetch: PropTypes.func.isRequired,
  round: PropTypes.number.isRequired,
};

const mapStateToProps = state => ({
  questions: state.questions,
  round: state.game.round,
});

const mapDispatchToProps = dispatch => ({
  gameReset: () => dispatch(gameResetted()),
  questionsFetch: () => dispatch(questionsFetchSent()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Initialize);
