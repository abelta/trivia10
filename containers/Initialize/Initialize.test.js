import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Initialize from '.';

describe('Initialize', () => {
  describe('when questions have not been loaded', () => {
    const state = {
      game: {
        round: 0,
      },
      questions: [],
    };
    const store = configureStore()(state);
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Initialize />
        </Router>
      </Provider>,
    ).toJSON();

    it('renders properly', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when questions have been loaded', () => {
    const state = {
      game: {
        round: 0,
      },
      questions: [
        {
          category: 'History',
          difficulty: 'Hard',
          text: 'Japan was part of the Allied Powers during World War I.',
          correctAnswer: 'true',
        },
      ],
    };
    const store = configureStore()(state);
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Initialize />
        </Router>
      </Provider>,
    ).toJSON();

    it('renders properly', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
