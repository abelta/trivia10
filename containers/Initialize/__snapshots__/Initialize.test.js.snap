// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initialize when questions have been loaded renders properly 1`] = `null`;

exports[`Initialize when questions have not been loaded renders properly 1`] = `
<View
  animationOptions={Object {}}
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
  styleName="flexible"
  virtual={undefined}
>
  <ActivityIndicator
    animating={true}
    color="grey"
    hidesWhenStopped={true}
    size={200}
    style={Object {}}
  />
</View>
`;
